python/lsst/daf/butler/instrument_tests.py

```diff
@@ -0,0 +1,95 @@
+# This file is part of daf_butler.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Helpers for writing tests against subclassses of Instrument.
+
+These are not tests themselves, but can be subclassed (plus unittest.TestCase)
+to get a functional test of an Instrument.
+"""
+
+import abc
+import dataclasses
+import os
+
+from lsst.daf.butler.core import Registry
+from lsst.daf.butler.core.butlerConfig import ButlerConfig
```

Comment:
Better to import these directly from `lsst.daf.butler`; the existence of `core` and any modules below that is considered an implementation detail (and only symbols lifted to package scope from it are considered public).